src/utils/resources/vmi/utils/cloud-init.ts

```diff
@@ -1,23 +1,47 @@
+import { load } from 'js-yaml';
+
 import { V1VirtualMachineInstance } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { CLOUD_INIT_MISSING_USERNAME } from '@kubevirt-utils/components/Consoles/utils/constants';
+
+type CloudinitUserDataObject = {
+  user?: string;
+  password?: string;
+  passwd?: { users: { name: string }[] };
+};
 
 export const getCloudInitCredentials = (
   vmi: V1VirtualMachineInstance,
-): { user: string; password: string } => {
+): { users: { name?: string; password?: string }[] } => {
   const cloudInitVolume = vmi?.spec?.volumes?.find(
     (volume) => volume?.cloudInitNoCloud || volume?.cloudInitConfigDrive,
   );
+  const cloudInitDataSource =
+    cloudInitVolume?.cloudInitNoCloud || cloudInitVolume?.cloudInitConfigDrive;
 
-  const cloudInit = cloudInitVolume?.cloudInitNoCloud || cloudInitVolume?.cloudInitConfigDrive;
-
-  const cloudInitValuesStrings = cloudInit?.userData
-    ?.split('\n')
-    ?.filter((row) => !row?.includes('#cloud-config'));
+  try {
+    const userDataObject: CloudinitUserDataObject = load(cloudInitDataSource?.userData);
 
-  const password = cloudInitValuesStrings
-    ?.find((row) => row?.includes('password'))
-    ?.split(': ')?.[1];
+    if (userDataObject?.user || userDataObject?.password) {
+      return {
+        users: [
+          {
+            name: userDataObject?.user || CLOUD_INIT_MISSING_USERNAME,
+            password: userDataObject?.password,
+          },
+        ],
+      };
+    }
 
-  const user = cloudInitValuesStrings?.find((row) => row?.includes('user'))?.split(': ')?.[1];
+    if (userDataObject?.passwd?.users) {
+      return {
+        users: userDataObject?.passwd?.users?.map((userobject) => {
+          return { name: userobject?.name };
```

Comment:
i think we should also have here || CLOUD_INIT_MISSING_USERNAME

Reply:
I am not sure, I'd better not to do it, as the `users` array is optional in the `passwd` object, and also because `CLOUD_INIT_MISSING_USERNAME` was originally used for the case when the username was missing while the password was present - which is not the case for the `passwd` object.

Reply:
Is it possible to have only sshkey without a name? If so we will present a nullish value.

Reply:
I think it is not possible. If already some user present in `users` array, the `name` is required, the other stuff is optional:
https://coreos.github.io/ignition/configuration-v3_3/

Reply:
I tried to run a machine with the same YAML in the bug and just removed the name, as you can see, the machine is running without any issue, which means there isn't a check for name guaranteed. We should ad a default value in this case, as we will get a User name: with an empty value... 
<img width="1714" alt="image" src="https://user-images.githubusercontent.com/14824964/182796681-e0c8d79b-63f9-4ac1-80e2-e1f088e0045a.png">

Reply:
Interesting. I don't understand the reason of why this issue should make the VM not running normally. Maybe it does not matter because the user cannot login without proper credentials anyway. But I'd expect some check from the backend side once the user changes the yaml this way, because it definitely is not correct like this, as the `name` is required there according to the doc (or the doc is not correct?). It shouldn't be possible to have there ssh key without any username. Thank you very much for checking it!

To solve this, IMO we should ignore such setting in the UI and to display that there are no credentials provided, as we usually do. Or to add some new text that some credentials are there but not set correctly. This is a question on someone who knows perfectly, how exactly this part of the UI should work in such cases. Or the UX. And we should raise this issue of not checking the yaml properly also to backend. WDYT?

Reply:
Opening a bug for BE is always recommended :). Feel free to do so.

We can tackle it in 2 ways, as I see.
1. adding  userobject?.name || CLOUD_INIT_MISSING_USERNAME
2. on line 37, before running the map function over the array, run a filter function and remove all the "users" without names.

In both cases, we are guaranteed to have a name in place, so it results in removing the optional from the name field at line 14.